tools/clang/lib/SPIRV/LowerTypeVisitor.h

```diff
@@ -98,6 +98,10 @@ class LowerTypeVisitor : public Visitor {
   SpirvDebugTypeComposite *lowerDebugTypeComposite(
       const RecordType *structType, const SpirvType *spirvType,
       llvm::SmallVector<StructType::FieldInfo, 4> &fields, bool isResourceType);
+  SpirvDebugTypeComposite *
+  lowerDebugTypeComposite(const SpirvType *type,
```

Comment:
What is the difference between this function and the one on line 98?

Reply:
The Google C++ guide has some info about [function overloading](https://google.github.io/styleguide/cppguide.html#Function_Overloading). Here we have two functions with the same name without clear indication why one should be used and not the other. If this one is only used for cbuffers for example, you can change the name to indicate that.

Reply:
Line 98 was for `QualType` based composite type lowering and this is for `SpirvType` based composite type lowering.

For some cases like `cbuffer`, we do not have `QualType` for it like a `HybridType`.
I changed the function name and add a comment.

Reply:
So Spiregg goes through the AST, for a struct we either have a `QualType` or a `HybridStructType`.

so why is `lowerDebugTypeCompositeFromSpirvType` taking a `StructType` rather than a `HybridStructType`?